### DISCUSSION TITLE
June Release - New design  🎨, automation logs, single Docker image, Snowflake integration and more.

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#6648] [Announcements]
Today we are launching the Budibase June release ✨ – and yes, we've updated our design section!

Budibase is growing up - we’ve added automation logs, Snowflake integration, single Docker image deployment, and more. Along with the existing data, design, and automation features, Budibase can finally deliver on the **professional low-code platform** promise. We think you'll love it! Now, let's get to the new features:

## New design 🎨
Yep. Budibase has a new, streamlined design UI! The new UI will speed up the design process and make building user interfaces a lot easier. In our opinion, it looks beautiful and is a joy to use - we’re super proud of how it’s turned out.

![pika-2022-07-08T15_00_58 471Z](https://user-images.githubusercontent.com/49767913/178299446-6c0b35e0-c325-44dd-9225-d7f7ee19c5d2.png)

## Automation logs 🚚
Enjoying Budibase’s automations? If so, we built automation logs exactly for you. Automation logs are a list of logs that display whether an automation was successful or not - including relevant information to help with debugging.
![pika-2022-07-11T15_41_02 549Z](https://user-images.githubusercontent.com/49767913/178302900-f57eb287-31a1-4c01-8553-887f332ceab0.png)

## Single Docker image deployment 🍱
Due to popular demand, you can now run Budibase on a single, ARM-compatible, Docker image. This allows you to run Budibase on Raspberry PI, M1 Mac, Vercel, Azure app services, AWS ECS, GCP Cloud Run, and more. For deployment all you need is a simple docker run command then you're set! It’s also possible to set up a custom domain, HTTPS, and reverse proxy with a simple environment variable, for example:

`CUSTOM_DOMAIN=acme.org`

After pointing your domain to the server, Budibase will handle: 
- enabling HTTPS 
- automatic certificate renewal

To get started with the single docker image:

```
docker run -d -t -p 10000:80 -v /local/path/data:/data budibase/budibase:latest
```

![carbon (3)](https://user-images.githubusercontent.com/49767913/178305454-23f2ec83-b04b-4dc9-a20e-ab9f2eb26da5.png)

## Use SSO token and Current User in REST requests 👤
You can now reuse an SSO token to hit other internal APIs as the currently logged-in user. This also unlocks OAuth 2.0.

## Snowflake integration ❄️
Connect Budibase to your Snowflake database and build internal applications on top of your Snowflake data in minutes.

---

### Bug Fixes 🐛
- You can now export data as CSV or JSON 
- Filtering Budibase DB columns with spaces #6421 
- Overview tab, Edit Icon, and color change not getting saved #6057
- Dynamic filter not filtering when using an options data type #5413
- Public page/form not working #6314
- 'money' Data Type Incompatible with Form Number Field #5708
- Error exporting data from table component #5431
- SQL Server - Filter More Than / Less Than on int causes validation failure #6398
- Displaying data within a JSON column broken #5911
- @ symbol in environment password var causes docker app-service to fail #5962
- Unable to select a table when another table with the same name exists #4809

---

## Coming in the next release 🔮

- [ ] Audit logs
- [ ] Barcode/QR scanner

---

## Thank you to our community 🙌
Thank you to our community. We are ever so thankful for your feedback and assistance. We've recently surpassed 14,000 Github stars and 30,000 companies are now using Budibase which is a great triumph. If you have not starred us, please do - it means a lot. Cheers from Joe and the entire Budibase team!

**Comment by JayP718** [comment]
Awesome, every time a new release comes out it feels like Christmas :)

**Comment by MarkOSullivan94** [comment]
Stacked release! Love to see it 🎉

**Comment by poirazis** [comment]
Congrats on the new release!
I have a couple of questions though.
1 . I have a production installation, Self Hosted / Debian. Can we migrate from a multi image docker installation to a single image ?
Do we do this via the CLI or do we need to make a new installation from scratch and migrate our users / apps etc ?

2. Can i hit "Check for Updates" from inside budibase or will this break things in the existing multi image installation ?

**Reply by mike12345567** [reply]
Hi @poirazis - right now migrations are a little painful, we have a solution which will be out soon, but is not released yet - see https://github.com/Budibase/budibase/pull/6584.

As for updates, the check for updates button is safe to use, but if you have the CLI it is generally a better method of updating.

**Comment by arladmin** [comment]
Hi team, love the new UI!
Especially the access-level settings for Navigation links.

I do have some feedback after experiencing the Builder's new UI hands-on, about one thing in particular:

- Earlier, the component tree was always visible on the left. While, the component selection menu was on the top. This, turns out, is very convenient while designing. 
Now, i find myself switching between the 'component menu' and the 'component tree' several times within a minute. Also, losing track/sight of the component tree seems to have a detrimental effect on the designing process, as that's primarily how we interact with the components on the screen (in fact, sometimes certain components might even be hidden due to certain conditions and the tree is the only way of managing them).

So, my request would be to make it so that the component tree and the component menu are simultaneously available for interaction, whenever one's involved in the designing process.

Thanks

**Reply by joebudi** [reply]
Thanks for the feedback @arladmin . We have a few ideas around streamlining the process of adding and positioning components. First, we want this version to settle, then we will gather further feedback on how to improve it once users have lived with it for a while.

**Reply by xnohat** [reply]
I agreed,
Screens tree with component is useful to me for fast edit, copy components between screens. New design UI is a drawback from old UI

**Comment by luizeof** [comment]
Thanks 🔥

**Comment by snvgglebear** [comment]
_You can now reuse an SSO token to hit other internal APIs as the currently logged-in user. This also unlocks OAuth 2.0._
Is there documentation of this? Does this mean that external apps can access non-public API's? (like the ones the app uses internally)

**Reply by shogunpurple** [reply]
https://docs.budibase.com/docs/rest-authentication#using-sso-tokens-in-rest-requests

**Comment by Movindutb** [comment]
It's Awesome :fire:

**Comment by xnohat** [comment]
New UI with deprecated Layout section is a drawback to me.
With Layout I can reuse components "combo" in many screens.

New UI also bad in navigate between screens, hard to copy and paste component between screens

Budibase is platform for not only simple app with a few screen, it can be a complex app with hundreds of screen (like my app). Original Design UI of Budibase are very easy to use for designing app, I hope a way to bring this UI back.

Budibase should more focus in stability of platform (functional bugs fixing) more than adding more features too fast but platform bugs prevent user can using Budibase in production.

I have choose Budibase to our company for internal applications developement, it help so much, but I also need spend many resources to fixing bug of Budibase and very scare when update new version of Budibase, because new version can break some thing still running in previous version.

Thanks team for all new updates

**Reply by developer-uhwi** [reply]
I agree with the layouts, it was a convenient of building multiple pages with common components, now this is very hard to do with the current UI.

Currently being forced to use one layout for all pages. In previous release, I was able to have a different navigation links based on the module I was in e.g separate administration navigation vs general navigation.

How do I implement such a scenario now, seems I will have to build custom navigation, hide the default navigation on pages, and copy and paste the custom navigation to the pages, where it will be located. 

So if I have a custom nav on 10 pages, and I change this custom nav, I will have to edit the nav on all 10 pages. 
Not cool Budibase.

Custom Layouts are really good for defining separate modules in your applications. 

What to do now????

**Reply by joebudi** [reply]
Hey! We've taken your feedback onboard, and we are working on a way to create a better 'layouts' experience. Layouts, in their old format, were creating more problems than they were solving, which is why we stripped them back. We will begin to add some functionality back as we learn more about what users need.

**Comment by sanskritkm** [comment]
M waiting for QR CODE 👯

**Reply by joebudi** [reply]
Hey @williamgh2019 

The Barcode/QR scanner was deprioritized in favor of a much larger, highly requested feature: custom components. The Barcode scanner/QR code reader will come and may come with or as a result of custom components.

**Comment by ryzam** [comment]
Hi @joebudi 
I really wanted to see professional look and feel for navigation menu, and I think just some extra effort from your team to add navigation menu with support added pick icon button or images

![image](https://user-images.githubusercontent.com/597759/182146002-09cb8798-38b6-4905-ab7d-6d9b489112b0.png)

**Reply by joebudi** [reply]
Thank you for your feedback @ryzam . I'll feed this back to the team.

**Comment by aborniak** [comment]
Hello Guys, 

>Budibase will handle -> enabling HTTPS, automatic certificate renewal

I am not sure how it works in Docker, but this doesn't work in Podman. 
Certbot uses HTTP-01 challenge - which means that 80 port should be up on the host, not inside container.
`podman run  -p 1000:80`  redirects port 1000 on host to port 80 in a container, but letsencrypt asking a host on port 80.


```
[root@aborniakFC ~]# nc -l 80
GET /.well-known/acme-challenge/chellange_key HTTP/1.1
Host: some-host.com
User-Agent: Mozilla/5.0 (compatible; Let's Encrypt validation server; +https://www.letsencrypt.org)
Accept: */*
Accept-Encoding: gzip
Connection: close
[root@aborniakFC ~]#
```

It fails inside the container.

```
root@57feb97d12a6:/# /app/letsencrypt/certificate-request.sh fc.my-clouds.net
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator webroot, Installer None
Requesting a certificate for some-host.com
Performing the following challenges:
http-01 challenge for some-host.com
Using the webroot path /var/www/html for all unmatched domains.
Waiting for verification...
Challenge failed for domain some-host.com
http-01 challenge for some-host.com
Cleaning up challenges
Some challenges have failed.

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: some-host.com
   Type:   connection
   Detail: 1.2.3.4: Fetching
   http://some-host.com/.well-known/acme-challenge/chellange_key:
   Connection refused

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address. Additionally, please check that
   your computer has a publicly routable IP address and that no
   firewalls are preventing the server from communicating with the
   client. If you're using the webroot plugin, you should also verify
   that you are serving files from the webroot path you provided.
ERROR: certbot request failed for some-host.com use http on port 80 - exiting
root@57feb97d12a6:/#

```
